src/relay/backend/build_module.cc

```diff
@@ -437,28 +441,51 @@ class RelayBuildModule : public runtime::ModuleNode {
     ret_.params = graph_codegen_->GetParams();
 
     auto lowered_funcs = graph_codegen_->GetLoweredFunc();
+
+    // When there is no lowered_funcs due to reasons such as optimization,
+    // we first try to generate a dummy one if the target host is "llvm".
     if (lowered_funcs.size() == 0) {
-      LOG(WARNING) << "no lowered funcs exist in the compiled module";
+      // Decide first the target host
+      Target target_host_val = target_host_;
+      if (!target_host_.defined()) {
+        for (const auto &it : targets_) {
+          if (it.second->device_type == kDLCPU) {
+            target_host_val = it.second;
+            break;
+          }
+        }
+      }
+
+      // If no target_host has been set, we choose a default one, which is
+      // llvm if "codegen.build_llvm" is accessible.
+      const runtime::PackedFunc* pf = runtime::Registry::Get("codegen.build_llvm");
+      if (!target_host_val.defined())
+        target_host_val = (pf != nullptr) ? target::llvm() : target::stackvm();
+
+      if (target_host_val.defined() && target_host_val->target_name == "llvm")
+        lowered_funcs.Set(
+          target_host_val->str(),
+          Array<LoweredFunc>({
+             MakeAPI(EvaluateNode::make(0), "__dummy__", Array<ObjectRef>(), 0, false) }));
```

Comment:
Add one `__dummy__` is not elegant solution IMO. If we have this LLVM IR:
```
target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-apple-macosx10.14.0"
```
Does this satisfy your need?

If it is, I think we could create one packed function in `llvm_module.cc`. We could create this kind of empty llvm module. Then we could have`ret_.mod = tvm::codegen::LLVMModuleCreate...` like CSourceModuleCreate.

Reply:
This sounds like a better way to go. I can try this out later.